portfolio/src/main/webapp/script.js

```diff
@@ -129,24 +129,73 @@ function loadComments() {
   }
   const fetchString = `/data?maxcomments=${maxcomments}&metric=${sortMetric}&order=${sortOrder}`;
   fetch(fetchString).then(response => response.json()).then(comments => {
-    const commentList = document.getElementById("comments");
+    const commentList = document.getElementById("toplevelcomments");
     while (commentList.lastChild) {
       commentList.removeChild(commentList.lastChild);
     }
-    for (const comment of comments) {
-      commentList.appendChild(createListElement(comment));
+    const commentTree = locateChildren(comments);
+    let numDisplayed = 0;
+    for (commentId in commentTree) {
+      if (numDisplayed == maxcomments) {
+        break;
+      }
+      let comment = commentTree[commentId];
+      if (comment["parentId"] === 0) {
+        numDisplayed++;
+        commentList.appendChild(constructReplyTree(comment, commentTree, 40));
```

Comment:
why is this 40 instead of 20?

Reply:
It represents the margin for replies to top-level comments. The margin for top-level comments is assigned to be 20 below and the margin increases by 20px for each level.